### PR TITLE
Fix glob resolver stalling on non-existent paths

### PR DESCRIPTION
**Pull request by PawelLipski**
Fix for #324

### CODE REVIEW

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/DependencyResolver.scala [hunk 1]
```diff
@@ -3,3 +3,9 @@ package org.virtuslab.ideprobe.dependencies
 trait DependencyResolver[Key] {
   def resolve(key: Key): Dependency
 }
+
```
Comment (lukaszwawrzyk) [inline core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/DependencyResolver.scala:6]: Why is this needed? We can simply pass function where such trait is expected anyway.

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala [hunk 2]
```diff
@@ -43,54 +43,40 @@ case class IntelliJPatternResolver(pattern: String) extends IntelliJResolver {
...
-    val replaced =
-      if (replacedBeforeResolvingGlobs.startsWith("file:"))
-        resolveGlobsInPattern(replacedBeforeResolvingGlobs)
-      else
-        replacedBeforeResolvingGlobs
 
-    Dependency(replaced)
+    if (artifactWithReplacementsApplied.startsWith("file:"))
+      findFilesMatchingGlob(artifactWithReplacementsApplied.stripPrefix("file:"))
+        .map("file:" + _)
```
Comment (lukaszwawrzyk) [inline core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala:52]: shouldn't we stript `file://`?

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/DependencyResolver.scala (hunk 1 above)
Comment (PawelLipski) [reply core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/DependencyResolver.scala:6]: Isn't needed per se... but makes it much more clear than we construct an instance of `DependencyResolver` 😅 

I was confused when saw

```scala
def resolver(artifact: String): DependencyResolver[IntelliJVersion] = { version =>
....
  Dependency(artifactWithReplacementsApplied)
}
```

Didn't notice the `version =>` and though that `def resolver` returns a `Dependency`, not a `DependencyResolver`

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala (hunk 2 above)
Comment (PawelLipski) [reply core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala:52]: Unfortunately, even the URIs returned by `java.lang.Class#getResource` (like, `java.lang.Class#getResource`) are only prefixed with `file:` instead of `file://`... so we probably shouldn't rely on standard being observed here 😅
